kbase-extension/static/kbase/js/widgets/narrative_core/upload/importErrors.js

```diff
@@ -0,0 +1,246 @@
+         * }]
+         * to:
+         * {
+         *   'file1.xls tab "tab_1"': [
+         *     'wrong column count, missing <some column>',
+         *     'duplicate specification found in file2.csv'
+         *   ],
+         *   'file2.csv': [
+         *     'duplicate specification found in file1.xls tab "tab_1"'
+         *   ],
+         * }
+         */
+        _processErrors() {
+            // a little helper to handle file error formatting
+            const addFileError = (error) => {
+                let fileKey = error.file;
+                if (error.tab) {
+                    fileKey += ` tab "${error.tab}"`;
+                }
+                if (!this.fileErrors[fileKey]) {
+                    this.fileErrors[fileKey] = [];
+                }
+                this.fileErrors[fileKey].push(error.message);
+            };
+
+            this.errors.forEach((error) => {
+                switch (error.type) {
+                    case BULK_SPEC_ERRORS.CANNOT_PARSE:
+                    case BULK_SPEC_ERRORS.INCORRECT_COLUMN_COUNT:
+                        addFileError(error);
+                        break;
+                    case BULK_SPEC_ERRORS.NOT_FOUND:
+                        addFileError(Object.assign({ message: 'File not found' }, error));
+                        break;
+                    case BULK_SPEC_ERRORS.MULTIPLE_SPECS:
+                        addFileError({
+                            file: error.file_1,
+                            tab: error.tab_1,
+                            message: error.message,
+                        });
+                        addFileError({
+                            file: error.file_2,
+                            tab: error.tab_2,
+                            message: error.message,
+                        });
+                        break;
+                    case BULK_SPEC_ERRORS.NO_FILES:
+                        this.noFileError = true;
+                        break;
+                    case BULK_SPEC_ERRORS.UNKNOWN:
+                        if (error.file) {
+                            addFileError(error);
+                        } else {
+                            this.unexpectedErrors.push(
+                                error.message || 'An unexpected error occurred!'
+                            );
+                        }
+                        break;
+                    case BULK_SPEC_ERRORS.SERVER:
+                        this.serverErrors.push(error.message);
+                        break;
+                    default:
+                        if (error.message) {
+                            this.unexpectedErrors.push(error.message);
+                        } else if (error.type) {
+                            this.unexpectedErrors.push(`Unknown error of type "${error.type}"`);
+                        } else {
+                            this.unexpectedErrors.push(`An unknown error occurred!`);
```

Comment:
this comment refers just to lines 141 and 154: is there a reason for having different wording ("unknown error" vs "unexpected error")? If not, do you want to use the same message for both? Alternatively, would it be worth altering the text of the first message to mention that it's an unexpected / unknown error from the staging service?

Reply:
I went back and forth in my head about that, and maybe was unclear in the end.
"unexpected" errors get returned from the server as an "unexpected_error" type.
I added "unknown" in the off chance that something creates an error without a type at all. But maybe I'm being too picky about edge cases?

Reply:
The default could just be "unexpected" error, and we use that label whenever something happens that doesn't fit under any other label, including an error without a label at all.